aten/src/ATen/cpu/vec256/vec256_complex_double.h

```diff
@@ -129,6 +129,18 @@ template <> class Vec256<std::complex<double>> {
     auto angle = _mm256_permute_pd(angle_(), 0x05); // angle    90-angle
     return _mm256_and_pd(angle, real_mask);         // angle    0
   }
+  __m256d polar_() const {
```

Comment:
The polar_() method is the key take-away here. It is essentially a private method for the Vec256 class. 

Now we have a decision to make:
1. We could modify the existing `operator*` and `operator/` to convert to polar coordinates, perform the op, and then convert to back to cartesian coordinates. I think you will find that we waste too much time converting between polar and cart for a single math kernel. Note: You lose considerable performance by calling multiple AVX functions.
2. Therefore, we might need to create a new dtype `ComplexPolarFloat` dtype so that we can keep the data in polar coordinates for multiple consecutive math ops. This will require significantly more work, but you are getting there.

Method 2 implies that we can have a different input dtype than the output dtype. PyTorch traditionally isn't setup to do this, but it has been attempted:
    - People have tried to cast complex to float for the `real(), imag(), abs(), angle()` kernels. This is hard because its also changes the memory size.
    - [le kernel](https://github.com/pytorch/pytorch/blob/4529d039711926c7388b83b1fcce2ec563fe3019/aten/src/ATen/native/cpu/BinaryOpsKernel.cpp#L355) successfully casts the output `byte` data type to `boolean` (both are 1 byte in memory).
    - #35524 is implementing the `std::complex<T>` data type internally in PyTorch. You could copy this code and create a `std::complex_polar<T>` C++ data type. Then you could do the same thing as the `le_kernel` and convert from `std::complex<T>` to `std::complex_polar<T>` when calling `polar()`

Reply:
In method 2 would we also need to add  `Vec256<std::complex_polar<T>>`?  I will look into `le` kernel, thanks for link.